Solutions/STM32F4DISCOVERY/STM32F4DISCOVERY.settings

```diff
@@ -16,6 +16,7 @@
     <IsSolutionWizardVisible>True</IsSolutionWizardVisible>
     <ENDIANNESS>le</ENDIANNESS>
     <NO_BOOTLOADER_COMPRESSION>true</NO_BOOTLOADER_COMPRESSION>
+    <PLATFORM_EMULATED_FLOATINGPOINT Condition="'$(COMPILER_TOOL)'=='GCC'">true</PLATFORM_EMULATED_FLOATINGPOINT>

```

Comment:
I understand this is required to enable use of newlib-nano, but is it really worth it to use emulation on a CPU with a hardware FPU just to save a few percents of the firmware size? 

ER_FLASH size built with GCC 5.4.1 and standard newlib is 327084, which is about 6% larger than 308380 reported in issue #474; the practical difference is even smaller due to the block size in the flash memory code area.

How about moving this setting to a separate 'size-optimized' solution?